src/contracts/libraries/BeaconChainProofs.sol

```diff
@@ -272,29 +278,32 @@ library BeaconChainProofs {
         require(proofs.timestampProof.length == 32 * (EXECUTION_PAYLOAD_HEADER_FIELD_TREE_HEIGHT),
             "BeaconChainProofs.verifyWithdrawalProofs: timestampProof has incorrect length");
 
-        {
-            /**
-            * Computes the block_header_index relative to the beaconStateRoot.  It concatenates the indexes of all the
-            * intermediate root indexes from the bottom of the sub trees (the block header container) to the top of the tree
-            */
-            uint256 blockHeaderIndex = BLOCK_ROOTS_INDEX << (BLOCK_ROOTS_TREE_HEIGHT)  | uint256(proofs.blockHeaderRootIndex);
-            // Verify the blockHeaderRoot against the beaconStateRoot
-            require(Merkle.verifyInclusionSha256(proofs.blockHeaderProof, beaconStateRoot, proofs.blockHeaderRoot, blockHeaderIndex),
-                "BeaconChainProofs.verifyWithdrawalProofs: Invalid block header merkle proof");
-        }
+
+        require(proofs.historicalSummaryBlockRootProof.length == 32 * (BEACON_STATE_FIELD_TREE_HEIGHT + (HISTORICAL_SUMMARIES_TREE_HEIGHT + 1) + 1 + (BLOCK_ROOTS_TREE_HEIGHT)),
+        "BeaconChainProofs.verifyWithdrawalProofs: historicalSummaryBlockRootProof has incorrect length");
+        /**
+        * Note: Here, the "1" in "1 + (BLOCK_ROOTS_TREE_HEIGHT)" signifies that extra step of choosing the "block_root_summary" within the individual 
+        * "historical_summary". Everywhere else it signifies merkelize_with_mixin, where the length of an array is hashed with the root of the array,
+        * but not here.
+        */
+        uint256 historicalBlockHeaderIndex = HISTORICAL_SUMMARIES_INDEX << ((HISTORICAL_SUMMARIES_TREE_HEIGHT + 1) + 1 + (BLOCK_ROOTS_TREE_HEIGHT)) | 
+                                            uint256(proofs.historicalSummaryIndex) << 1 + (BLOCK_ROOTS_TREE_HEIGHT) |
+                                            BLOCK_SUMMARY_ROOT_INDEX << (BLOCK_ROOTS_TREE_HEIGHT) | uint256(proofs.blockHeaderRootIndex);
```

Comment:
I'm trying to make sure I have the order of operations this is doing correct, and think some parentheses might be a good idea here.
Let me know if this is correct / here's a suggestion for what I personally think is clearer in terms of order-of-operations:
```
        uint256 historicalBlockHeaderIndex =
               (HISTORICAL_SUMMARIES_INDEX << ((HISTORICAL_SUMMARIES_TREE_HEIGHT + 1) + 1 + (BLOCK_ROOTS_TREE_HEIGHT))) | 
               (uint256(proofs.historicalSummaryIndex) << (1 + BLOCK_ROOTS_TREE_HEIGHT)) |
               (BLOCK_SUMMARY_ROOT_INDEX << BLOCK_ROOTS_TREE_HEIGHT) | 
               uint256(proofs.blockHeaderRootIndex);
```

Reply:
if Im reading this correct, each of the components in between the bitwise or's are wrapped in parenthesis?  Yeah that looks cleaner, added it